tests/cpp/test_pdl.cpp

```diff
@@ -0,0 +1,109 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <vector>
+
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <ops/all_ops.h>
+#include <scheduler/tools/inlining.h>
+#include <tests/cpp/utils.h>
+#include <tests/cpp/validator.h>
+
+namespace nvfuser {
+
+using testing::_;
+using testing::ElementsAre;
+
+class ProgrammaticDependentLaunchTest : public NVFuserTest {
+ protected:
+  void SetUp() override {
+    NVFuserTest::SetUp();
+    NVFUSER_TEST_CUDA_ARCH_GUARD(9, 0);
+  }
+};
+
+TEST_F(ProgrammaticDependentLaunchTest, Basic) {
+  std::shared_ptr<Fusion> fusion_ptr = std::make_shared<Fusion>();
+  Fusion& fusion = *fusion_ptr;
+  FusionGuard fg(&fusion);
+
+  TensorView* tv0 = makeContigTensor(2);
+  TensorView* tv1 = makeContigTensor(2);
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+
+  TensorView* tv2 = add(tv0, tv1);
+  fusion.addOutput(tv2);
+
+  // Clone prescheduled fusion for validation
+  std::shared_ptr<Fusion> presched_fusion = std::make_shared<Fusion>(fusion);
+
+  // Cache the inputs and output
+  TensorView* tv0_cached = tv0->cacheAfter();
+  TensorView* tv1_cached = tv1->cacheAfter();
+  TensorView* tv2_cached = tv2->cacheBefore();
+
+  TensorView* grid_wait = wait_for_prior_grid({tv0, tv1});
+  tv0_cached->addDependency(grid_wait);
+  tv1_cached->addDependency(grid_wait);
+
+  TensorView* grid_launch = launch_dependent_grid({tv2_cached});
+  tv2->addDependency(grid_launch);
```

Comment:
IIUC, this would put the PDL notification before the store of tv2. Is there any way to put the notification at the very end of the kernel?